url24d793285d_auzll_gengweibo_tgz-pJ8-org_json_TestJ8/src/com/gengweibo/weibo/WeiboSohu.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * http://auzll.iteye.com/
+ */
+package com.gengweibo.weibo;
+
+import static com.gengweibo.weibo.Weibo.Method.GET;
+import static com.gengweibo.weibo.Weibo.Method.POST;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * @author auzll@msn.com
+ * @since 2011-3-16
+ */
+public class WeiboSohu extends Weibo implements IWeibo {
+
+    /**  */
+    private static final long serialVersionUID = 3375792387585554576L;
+
+    public WeiboSohu() {
+        super("http://api.t.sohu.com/oauth/request_token",
+                "http://api.t.sohu.com/oauth/authorize",
+                "http://api.t.sohu.com/oauth/access_token",
+                "http://api.t.sohu.com", OAuthProperty.getCallbackUrl(),
+                OAuthProperty.CONSUMER_KEY_SOHU,
+                OAuthProperty.CONSUMER_SECRET_SOHU, WeiboType.T_SOHU);
+    }
+
+    public void bindWeiboAccountContext() {
+        if (null == weiboId && null == weiboAccountName) {
+            Response response = verifyCredentials();
+            try {
+                JSONObject jsonObject = new JSONObject(
+                        response.readBodyAsString());
+                weiboId = getType().getEnName() + "_"
+                        + jsonObject.getString("id");
+                weiboAccountName = jsonObject.getString("screen_name");
+                if (null != weiboAccountName) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It seems unusual to set the value of a variable only when it is already set. You might want to switch the null check of != on line 39 to == check before setting the field value on line 40 or verify if the check is needed at all.